scalyr_agent/log_processing.py

```diff
@@ -2877,11 +2877,29 @@ def process_line(self, input_line):
 
         modified_it = False
 
-        for redaction_rule in self.__redaction_rules:
-            (input_line, redaction) = self.__apply_redaction_rule(
-                input_line, redaction_rule
-            )
-            modified_it = modified_it or redaction
+        try:
+            for redaction_rule in self.__redaction_rules:
+                (input_line, redaction) = self.__apply_redaction_rule(
+                    input_line, redaction_rule
+                )
+                modified_it = modified_it or redaction
+        except re.error as e:
+            if e.message == "unmatched group":  # pylint: disable=no-member
+                log.error(
+                    "Error while getting the default gateway: %s. Please make sure any redaction rules only reference groups that are guaranteed to match.",
```

Comment:
Default gateway?  I think you mean something like "Error while executing redaction rules".

Also, let's include the text of the redaction rule as well here.

Reply:
I have no idea how that got there...